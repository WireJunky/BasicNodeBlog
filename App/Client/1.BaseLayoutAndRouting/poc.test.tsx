import * as React from 'react';

const STATUS = {
  HOVERED: 'hovered',
  NORMAL: 'normal',
};

export default class Link extends React.Component {

  constructor(props) {
    super(props);

    this._onMouseEnter = this._onMouseEnter.bind(this);
    this._onMouseLeave = this._onMouseLeave.bind(this);

    this.state = {
      class: STATUS.NORMAL,
    };
  }

  _onMouseEnter() {
    this.setState({class: STATUS.HOVERED});
  }

  _onMouseLeave() {
    this.setState({class: STATUS.NORMAL});
  }

  render() {
    return (
      <a
        href={'#'}
        onMouseEnter={this._onMouseEnter}
        onMouseLeave={this._onMouseLeave}>
        {this.props.children}
        <div></div>
      </a>
    );
  }

}

import * as renderer from 'react-test-renderer';

describe('Link changes the class when hovered', () => {
    it('trial', () => {
        const component = renderer.create(
          <Link>Facebook</Link>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        // manually trigger the callback
        tree.props.onMouseEnter();
        // re-rendering
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        
        // manually trigger the callback
        tree.props.onMouseLeave();
        // re-rendering
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();
        }); 
});   
