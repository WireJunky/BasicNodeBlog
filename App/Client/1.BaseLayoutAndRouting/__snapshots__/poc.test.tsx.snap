// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered trial 1`] = `
<a
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
  <a />
</a>
`;

exports[`Link changes the class when hovered trial 2`] = `
<a
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
  <a />
</a>
`;

exports[`Link changes the class when hovered trial 3`] = `
<a
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
  <a />
</a>
`;
